brave/brave-core
# Android P3A

Resolves https://github.com/brave/brave-browser/issues/6176

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (unit, browser, security tests) on
- [not sure] Added appropriate QA labels (`QA/Yes` or `QA/No`) to the associated issue
- [not sure] Added appropriate release note labels (`release-notes/include` or `release-notes/exclude`) to the associated issue

## Test Plan:

I. Clean install
1. Install this PR apk as a clean installation
2. Launch, ensure you can't see informers
3. Go to Settings => Advanced => Privacy, ensure `Automatically send completely private product analytics to Brave` is unchecked
4. Restart app, Go to Settings => Advanced => Privacy, ensure `Automatically send completely private product analytics to Brave` is still unchecked 
5. Check `Automatically send completely...` checkbox,
6. Restart app, ensure `Automatically send completely...` is still checked

II. Update install
1. Install version which does not contain this PR, for example, 1.18.45
2. Repeat all the steps from (I)

Advanced test to ensure PR is ready for onboarding changes, requires code change between runs, is not possible to perform by QA team 

=====

0. This require to prepare special apk
comment out 
```
            if (!BravePrefServiceBridge.getInstance().hasPathP3AEnabled()) {
                BravePrefServiceBridge.getInstance().setP3AEnabled(false);
            }
```
and uncomment 
```
            if (!PackageUtils.isFirstInstall(this)) {
                BraveP3AInformers.show();
            } else {
                // On the first run P3A is must be set from the onboarding wizard
            }
```
at  `src/brave/android/java/org/chromium/chrome/browser/app/BraveActivity.java`

III. Ready for onboarding, clean install 
1. Install modified apk in a clean way
2. Run, ensure there is no informer displayed
3. Go to Settings => Advanced => Privacy, ensure `Automatically send completely private...` is checked
(P3A should be disabled or kept enabled by onboarding wizard, follow up issue)

IV. Ready for onboarding, update install 
1. Install version which does not contain this PR, for example, 1.18.45
2. Run and exit to go through onboarding
3. Install modified apk
4. Run, ensure informer is shown
5.1 press `Learn More` ensure page `https://brave.com/P3A` is opened

5.2 repeat 5.1-5.4, press cross, ensure `Automatically send completely private...` is checked
5.3 repeat 5.1-5.4, press `Disable`, ensure `Automatically send completely private...` is unchecked
5.4 repeat 5.1-5.4, press `Got it`, ensure `Automatically send completely private...` is checked

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on.
